File: python3/number_of_1_bits.py
"""
Space   : O(1)
Time    : O(n)
"""


class Solution:
    def hammingWeight(self, n: int) -> int:
        ans = 0
        while n != 0:
            if n & 1 == 1:
                ans += 1
            n >>= 1
        return ans

File: golang/number_of_1_bits.go
package main

/*
Space   : O(1)
Time    : O(n)
*/

func hammingWeight(num uint32) int {
	ans := 0

	for num > 0 {
		if num%2 == 1 {
			ans++
		}
		num /= 2
	}

	return ans
}
